pkg/api/kubegraph/analysis/hpa.go

```diff
@@ -0,0 +1,66 @@
+package analysis
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	kubegraph "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+)
+
+const (
+	HPAMissingCPUTargetError      = "HPAMissingCPUTarget"
+	HPAOverlappingScaleRefWarning = "HPAOverlappingScaleRef"
+)
+
+func FindHPASpecsMissingCPUTargets(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, uncastNode := range g.NodesByKind(kubegraph.HorizontalPodAutoscalerKind) {
+		node := uncastNode.(*kubegraph.HorizontalPodAutoscalerNode)
+
+		hpa, ok := node.Object().(*extensions.HorizontalPodAutoscaler)
+		if !ok {
+			continue
+		}
+
+		if hpa.Spec.CPUUtilization == nil {
+			markers = append(markers, osgraph.Marker{
+				Node:     node,
+				Severity: osgraph.ErrorSeverity,
+				Key:      HPAMissingCPUTargetError,
+				Message:  fmt.Sprintf("horizontal pod autoscaler %q is missing a CPU utilization target", hpa.Name),
+			})
+		}
+	}
+
+	return markers
+}
+
+func FindOverlappingHPAs(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+	references := map[extensions.SubresourceReference]string{}
+
+	for _, uncastNode := range g.NodesByKind(kubegraph.HorizontalPodAutoscalerKind) {
+		node := uncastNode.(*kubegraph.HorizontalPodAutoscalerNode)
+
+		hpa, ok := node.Object().(*extensions.HorizontalPodAutoscaler)
+		if !ok {
+			continue
+		}
+
+		if otherHPA, ok := references[hpa.Spec.ScaleRef]; ok {
```

Comment:
> Which HPA node will we tie the warning marker to? Or should we make a marker for each one?

RCs create a marker on each node.  That seems reasonable and it allows good filtering by Node.